### PR TITLE
Enable VERIFY_CERTIFICATE_AT_CLIENT by default

### PR DESCRIPTION
**Pull request by Kmoneal**
Enable proxies using their OS CA certificate bundle by default.

Enable `VERIFY_CERTIFICATE_AT_CLIENT` by default.

### CODE REVIEW

**Comment by Kmoneal** [comment]
@kfaseela

**Review by howardjohn — COMMENTED** [review]
I recall in the design doc there was some roadmap to make this safer. Did those items get merged?

**Comment by Kmoneal** [comment]
> I recall in the design doc there was some roadmap to make this safer. Did those items get merged?

Not to my knowledge but I will double check with Faseela and look at next steps

**Comment by kfaseela** [comment]
> > I recall in the design doc there was some roadmap to make this safer. Did those items get merged?
> 
> Not to my knowledge but I will double check with Faseela and look at next steps

May be the PR can be updated with the links to whatever items were already done and merged. And what is pending. Atleast I see that the updates to security best practices documentation and other similar items have been merged.

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -597,7 +597,7 @@ var (
 		"If false, TCP probes will not be rewritten and therefor always succeed when a sidecar is used.",
 	).Get()
 
-	EnableQUICListeners = env.RegisterBoolVar("PILOT_ENABLE_QUIC_LISTENERS", false,
```
Comment (kfaseela) [inline pilot/pkg/features/pilot.go:600]: Aren't we supposed to change VERIFY_CERTIFICATE_AT_CLIENT default value?

Comment (Kmoneal) [reply pilot/pkg/features/pilot.go]: My bad, just pushed up the correction

**Comment by Kmoneal** [comment]
- [X] Add VERIFY_CERTIFICATE_AT_CLIENT as an opt-in environment variable
- [ ] Added logging to warn users of proxies that are not verifying host certificates
- [ ] Make VERIFY_CERTIFICATE_AT_CLIENT an opt-out environment variable
- [ ] Add Metrics to warn of connections to external sites that lack `caCertificate`s
- [ ] Add Analyzer to warn of external hosts being connected to without certificate verification

**Comment by Kmoneal** [comment]
We don't have the metrics. Is there value in adding metrics? Do we know if users will benefit from metrics?

**Comment by kfaseela** [comment]
Hello @Kmoneal @jacob-delgado  ,  will you be able to list down the changes needed at the integration test level to get this working? I remember you had mentioned this to me sometime back, but I have lost it. Better to add it here for tracking?

**Comment by kfaseela** [comment]
This is covered by https://github.com/istio/istio/pull/48225

**Pull request closed without merging**